### PR TITLE
Store item sort/filter

### LINKED ISSUE #43: New arrivals
Sort new store items (<1w) at the top of the list w/ badge

### LINKED ISSUE #44: Filter store to only what I can afford

### LINKED ISSUE #61: Store item sorting and filtering

### PR DESCRIPTION
**Pull request by asportnoy**
- Add new arrivals for store items (Less than 1 week old). Will show a "new" badge next to the name.
- Add option to sort store items. Defaults to alphabetical. New arrivals will be sorted first except for date-related sort options
- Add option to search store items
- Add option to filter store to only the items the viewer can afford. Will only be visible to students.

### CODE REVIEW

File: frontend/src/pages/store/[store].svelte [hunk 1]
```diff
@@ -479,28 +520,66 @@
 				>
 			</div>
 		{/if}
-		{#if error || !items || items.length == 0}
-			<h2 class="text-center">
+		<div class="flex space-x-2 mb-4">
```
Comment (mswider) [inline frontend/src/pages/store/[store].svelte]: Right now these feel really close together, you might want to make the spacing bigger to something like `space-x-4`

**Review by mswider — APPROVED** [review]
Looks good, I just had something with the filter spacing. 
Also, I noticed you're doing the new arrival indicator as a badge, which looks fine, but when I'm adding categories I was going to use badges as well. Do you have a way that we should differentiate between the categories and the new badge? Would we need to?

**Comment by asportnoy** [comment]
Go ahead and make the spacing change, I'm not at my computer so I can't do it right now.

The category can be below the description and in a different color. But even if we do put them in the same place it should be fine.

**Comment by mswider** [comment]
I've also just noticed that for all filters except for "oldest to newest", the new arrivals are always put first, even when it isn't correct. This should be fixed as it is not correct behavior.

**Comment by asportnoy** [comment]
This is intentional. New items are sorted first except for the date related filters.

**Comment by mswider** [comment]
> This is intentional. New items are sorted first except for the date related filters.

Why? If someone sorts something by price, wouldn't they expect the cheapest things to _always_ be first?

**Comment by asportnoy** [comment]
Should I have it only happen on name sort (default) or just not at all?

**Comment by asportnoy** [comment]
@mswider I added a "Featured" sort option which prioritizes new items and removed it from the other sorting methods. Once we implement #65 this will go on the featured sort as well.

**Comment by mswider** [comment]
> @mswider I added a "Featured" sort option which prioritizes new items and removed it from the other sorting methods. Once we implement #65 this will go on the featured sort as well.

Nicely done, that works

**Review by mswider — APPROVED** [review]

**Comment by mswider** [comment]
I also just noticed that the toggle is pretty invisible when it isn't checked in dark mode
image

**Comment by asportnoy** [comment]
Adding a collapse makes the page look cleaner overall. The background fixes the switch visibility issue.

**Review by mswider — APPROVED** [review]
Nice addition

**Pull request merged**